File: mobile/src/main/java/info/papdt/express/helper/support/HttpUtils.java
package info.papdt.express.helper.support;

import android.util.Log;

import java.io.IOException;

import info.papdt.express.helper.model.BaseMessage;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class HttpUtils {

	private static OkHttpClient client = new OkHttpClient();

	private static final String UA_CHROME = "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.84 Safari/537.36";

	private static final String TAG = HttpUtils.class.getSimpleName();

	public static BaseMessage<String> getString(String url, String ua) {
		BaseMessage<String> result = new BaseMessage<>();

		Request request = new Request.Builder().url(url).addHeader("User-Agent", ua).build();
		try {
			Response response = client.newCall(request).execute();
			result.setCode(response.code());
			result.setData(response.body().string());
			Log.i(TAG, result.getData());
		} catch (IOException e) {
			result.setCode(BaseMessage.CODE_ERROR);
			e.printStackTrace();
		}

		return result;
	}

	public static BaseMessage<String> getString(String url, boolean useLocalUA) {
		return getString(
				url,
				useLocalUA ? System.getProperty("http.agent") : UA_CHROME
		);
	}

}

File: mobile/src/main/kotlin/info/papdt/express/helper/support/HttpUtils.kt
package info.papdt.express.helper.support

import android.util.Log

import java.io.IOException
import java.util.concurrent.TimeUnit

import info.papdt.express.helper.model.BaseMessage
import okhttp3.OkHttpClient
import okhttp3.Request

object HttpUtils {

    private val client = OkHttpClient.Builder()
            .connectTimeout(10, TimeUnit.SECONDS)
            .readTimeout(10, TimeUnit.SECONDS)
            .build()

    private const val UA_CHROME = "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.84 Safari/537.36"

    private val TAG = HttpUtils::class.java.simpleName

    fun getString(url: String, ua: String): BaseMessage<String> {
        val result = BaseMessage<String>()

        val request = Request.Builder().url(url).addHeader("User-Agent", ua).build()
        try {
            val response = client.newCall(request).execute()
            result.code = response.code()
            result.data = response.body()!!.string()
            Log.i(TAG, result.data)
        } catch (e: IOException) {
            result.code = BaseMessage.CODE_ERROR
            e.printStackTrace()
        }

        return result
    }

    fun getString(url: String, useLocalUA: Boolean): BaseMessage<String>
            = getString(url, if (useLocalUA) System.getProperty("http.agent") else UA_CHROME)

}
